agent/api/containerstatus.go

```diff
@@ -22,6 +22,13 @@ const (
 	ContainerCreated
 	// ContainerRunning represents a container that has started
 	ContainerRunning
+	// ContainerResourcesProvisioned represents a container that has completed provisioning all of its
+	// resources. Non-internal containers (containers present in the task definition) transition to
```

Comment:
I'm a *little* concerned about non-internal containers going to this state; it's going to look very weird in logs to see non-internal containers in RESOURCES_PROVISIONED rather than RUNNING and I think we'll be creating confusion to do that.

Should we make it so that non-internal containers never actually transition to this state instead?  That would mean that the DesiredStatus (and SteadyStateStatus) for a non-internal container would be RUNNING but the DesiredStatus (and SteadyStateStatus) for an internal container would be RESOURCES_PROVISIONED.